src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -290,41 +292,42 @@ private boolean hasText(final DetailAST slistAST) {
         final int slistColNo = slistAST.getColumnNo();
         final int rcurlyLineNo = rcurlyAST.getLineNo();
         final int rcurlyColNo = rcurlyAST.getColumnNo();
-        final String[] lines = getLines();
         boolean returnValue = false;
         if (slistLineNo == rcurlyLineNo) {
             // Handle braces on the same line
-            final String txt = lines[slistLineNo - 1]
-                    .substring(slistColNo + 1, rcurlyColNo);
-            if (!CommonUtil.isBlank(txt)) {
+            final int[] txt = Arrays.copyOfRange(getLineCodePoints(slistLineNo - 1),
+                    slistColNo + 1, rcurlyColNo);
+
+            if (!CodePointUtil.isBlank(txt)) {
                 returnValue = true;
             }
         }
         else {
-            final String firstLine = lines[slistLineNo - 1].substring(slistColNo + 1);
-            final String lastLine = lines[rcurlyLineNo - 1].substring(0, rcurlyColNo);
+            final int[] codePointsFirst = getLineCodePoints(slistLineNo - 1);
```

Comment:
I think it would be good to keep previous naming convention, i.e. `codePointsFirstLine` and `codePointsLastLine`. @MUzairS15 what do you think?

Reply:
Yes, better.
Done

Reply:
Resolved.